src/core/reactor.ts

```diff
@@ -2587,36 +2581,12 @@ export class App extends Reactor {
     console.log(apg.toString());
 
     Log.debug(this, () => "Before collapse: " + apg.toString());
-    const collapsed = new SortableDependencyGraph();
 
     // 1. Collapse dependencies and weed out the ports.
-    const leafs = apg.leafNodes();
-    const visited = new Set();
-
-    function search(
-      reaction: Reaction<unknown>,
-      nodes: Set<Port<Present> | Reaction<unknown>>
-    ): void {
-      for (const node of nodes) {
-        if (node instanceof Reaction) {
-          collapsed.addEdge(reaction, node);
-          if (!visited.has(node)) {
-            visited.add(node);
-            search(node, apg.getEdges(node));
-          }
-        } else {
-          search(reaction, apg.getEdges(node));
-        }
-      }
-    }
-
-    for (const leaf of leafs) {
-      if (leaf instanceof Reaction) {
-        collapsed.addNode(leaf);
-        search(leaf, apg.getEdges(leaf));
-        visited.clear();
-      }
-    }
+    const collapsed = SortablePrecedenceGraph.fromPrecedenceGraph(
```

Comment:
Same here, the issue we encountered during our meeting today is reproducible in VSCode but eslint doesn't complain